hudi-client/src/main/java/org/apache/hudi/client/HoodieWriteClient.java

```diff
@@ -339,8 +352,13 @@ protected void postCommit(HoodieCommitMetadata metadata, String instantTime,
       archiveLog.archiveIfRequired(jsc);
       if (config.isAutoClean()) {
         // Call clean to cleanup if there is anything to cleanup after the commit,
-        LOG.info("Auto cleaning is enabled. Running cleaner now");
-        clean(instantTime);
+        if (config.isRunParallelAutoClean()) {
```

Comment:
let's move this into its own method for readability?

Reply:
Done